### PR TITLE
fix two bugs for mips64le

### PR DESCRIPTION
**Pull request by wanghuaiqing2010**
Signed-off-by: wanghuaiqing <wanghuaiqing@loongson.cn>

### CODE REVIEW

**Comment by wanghuaiqing2010** [comment]
@tonistiigi PTAL

File: scripts/cross.sh [hunk 1]
```diff
@@ -74,7 +74,7 @@ case "$TARGETARCH" in
   PKG_PREFIX="i686-linux-gnu"
   ;;
 "mips64le")
-  OUT_ARCH="mips64"
+  OUT_ARCH="mips64le"
```
Comment (tonistiigi) [inline scripts/cross.sh:77]: Are you sure this order is correct. Can you give me an example output of full uname

Comment (wanghuaiqing2010) [reply scripts/cross.sh:77]: if TARGET_ARCH is mips64le, OUT_ARCH="mips64" will cause the following problem.
QEMU_TARGETS will include mips64el-linux-user and not include mips64-linux-user.
This is not the result we want.

```
arch="$(cross.sh arch)"

if [ "$arch" != "x86_64" ]; then
    QEMU_TARGETS="$QEMU_TARGETS x86_64-linux-user"
fi
if [ "$arch" != "aarch64" ]; then
    QEMU_TARGETS="$QEMU_TARGETS aarch64-linux-user"
fi
if [ "$arch" != "armv7l" ] && [ "$arch" != "armv6l" ] ; then
    QEMU_TARGETS="$QEMU_TARGETS arm-linux-user"
fi
if [ "$arch" != "riscv64" ]; then
    QEMU_TARGETS="$QEMU_TARGETS riscv64-linux-user"
fi
if [ "$arch" != "ppc64le" ]; then
    QEMU_TARGETS="$QEMU_TARGETS ppc64le-linux-user"
fi
if [ "$arch" != "s390x" ] && [ "$arch" != "riscv64" ] ; then
    QEMU_TARGETS="$QEMU_TARGETS s390x-linux-user"
fi
if [ "$arch" != "i386" ] ; then
    QEMU_TARGETS="$QEMU_TARGETS i386-linux-user"
fi
if [ "$arch" != "mips64le" ] ; then
    QEMU_TARGETS="$QEMU_TARGETS mips64el-linux-user"
fi
if [ "$arch" != "mips64" ] ; then
    QEMU_TARGETS="$QEMU_TARGETS mips64-linux-user"
fi

if cross.sh is_cross; then
    FLAG_CROSS_PREFIX="--cross-prefix=$(cross.sh cross-prefix)-"
fi
```

Comment (tonistiigi) [reply scripts/cross.sh:77]: I agree that `mips` is wrong in here but wondered if `uname -m` is `mips64le` or `mips64el`

Comment (wanghuaiqing2010) [reply scripts/cross.sh:77]: Below is the result of my execution on Loongson's system.
The result is not what I expected.
But I think the arch of the cross-compiled host is usually amd64. 
Therefore, I don’t think it is very meaningful to judge the arch of the host. 


```
[root@localhost ~]# uname -a
Linux localhost.localdomain 3.10.84-23.fc21.loongson.8.mips64el #1 SMP PREEMPT Thu Mar 5 17:56:57 CST 2020 mips64 mips64 mips64 GNU/Linux
[root@localhost ~]# 
[root@localhost ~]# 
[root@localhost ~]# 
[root@localhost ~]# uname -m
mips64
[root@localhost ~]#
```

**Review by tonistiigi — APPROVED** [review]

File: scripts/cross.sh (hunk 1 above)
Comment (tonistiigi) [reply scripts/cross.sh:77]: for correctness then maybe it makes sense to manually detect the endianness of the system. And if it is LE then swap mips64 to mips64le/mips64el. More interested of this in https://github.com/tonistiigi/xx/pull/14 than in here though

Comment (wanghuaiqing2010) [reply scripts/cross.sh:77]: Yes, I agree with you.

**Comment by wanghuaiqing2010** [comment]
@tonistiigi could we merge this commit to master branch?

**Pull request merged**